src/core/trtserver.h

```diff
@@ -792,6 +810,28 @@ TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_ServerSharedMemoryAddress(
     TRTSERVER_Server* server, TRTSERVER_SharedMemoryBlock* shared_memory_block,
     size_t offset, size_t byte_size, void** base);
 
+/// \param server The inference server object.
+/// \param shared_memory_block The system shared memory block.
+/// \param offset The offset within the shared memory block to get the
+/// address for.
+/// \param byte_size The size of the memory block within the shared memory
+/// block. Returns error if a block of this size (starting at
+/// 'offset') isn't completely contained in the shared memory block.
+/// \param base Returns the base address.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_ServerCudaSharedMemoryAddress(
+    TRTSERVER_Server* server, TRTSERVER_SharedMemoryBlock* shared_memory_block,
+    size_t offset, size_t byte_size, void** cuda_base, size_t* cuda_byte_size);
+
+/// \param shared_memory_block The CUDA shared memory block.
+/// \param kind Returns the kind of device the shared memory region is in (CPU
+///  = 0, GPU = 1)
+/// \param device id Returns the GPU number the shared memory region is in.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_ServerSharedMemoryDevice(
```

Comment:
This doesn't belong here. It could be a TRTSERVER_SharedMemoryBlock... function but we don't have other such query function right now. Why is it needed?

Reply:
To know if the shared memory region is on the CPU/GPU and in case of GPU, which GPU.